src/Compilers/Core/Portable/InternalUtilities/Debug.cs

```diff
@@ -19,7 +19,7 @@ public static void Assert([DoesNotReturnIf(false)] bool b, string message)
             => Debug.Assert(b, message);
 
         [Conditional("DEBUG")]
-        public static void AssertNotNull<T>([NotNull] T value) where T : class?
```

Comment:
This was an expected constraint. Non-null assertions make no sense for value types. Why are we removing it?

📝 If we need support for nullable value types, an overload could be added.

Reply:
With the constraint the method can't be used for unconstrained generic type.

Reply:
For me, that is a desirable side effect. The sibling method `RoslynDebug.Assert` can already be used without unnecessarily relaxing the constraint here.

The same applies for using `ThrowIfFalse` instead of `ThrowIfNull`.

Reply:
We can do that.